src/main/java/org/example/hw5/src/main/java/org/example/repository/impl/UserRepositoryImpl.java

```diff
@@ -0,0 +1,51 @@
+package org.example.repository.impl;
+
+import org.example.model.User;
+import org.example.repository.UserRepository;
+import org.example.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+import java.util.List;
+import java.util.Optional;
+
+public class UserRepositoryImpl implements UserRepository {
+    private final SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+
+    @Override
+    public User create(User user) {
+        try (Session session = sessionFactory.openSession()) {
+            session.save(user);
+        }
+        return user;
+    }
+
+    @Override
+    public User update(User user) {
+        Transaction transaction = null;
+        try (Session session = sessionFactory.openSession()) {
+            transaction = session.getTransaction();
+            transaction.begin();
+            session.update(user);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+        return user;
+    }
+
+    @Override
+    public Optional<User> findById(Long id) {
+        return Optional.ofNullable(sessionFactory.openSession().find(User.class, id));
+    }
+
+    @Override
+    public List<User> findAll() {
+        return sessionFactory.openSession()
```

Comment:
it's better to use try-with-resourses construction and sessions will be closed automatically